sklearn/utils/tests/test_class_weight.py

```diff
@@ -117,6 +117,15 @@ def test_compute_class_weight_balanced_unordered():
     assert_almost_equal(np.dot(cw, class_counts), y.shape[0])
     assert_array_almost_equal(cw, [2., 1., 2. / 3])
 
+def test_class_weight_with_string_label():
+    y = np.asarray(["A","A","A","B","B","C"])
+    classes = np.unique(y)
+    class_weights = {c: 1.0 for c in classes}
+    class_weights['D'] = 1.0  # This should get a proper ValueError
+    cw = assert_raises(ValueError, compute_class_weight, class_weights, 

```

Comment:
Well done for going through your git/github initiation! There are a few ways you can simplify this test. One is to 
reuse `test_compute_class_weight_not_present` and add your test there rather than creating a new function after this [line](https://github.com/MMeketon/scikit-learn/blob/1dec75c18904b35ec147b7e1a0d9dbd80b089fbe/sklearn/utils/tests/test_class_weight.py#L33).

I think that just adding this should be enough:
```py
# Fix exception in error message formatting when missing label is a string
# https://github.com/scikit-learn/scikit-learn/issues/8312
assert_raise_message(ValueError,
                     'Class label label_not_present not present',
                     compute_class_weight,
                     {'label_not_present': 1.}, classes, y)
```